src/Sylius/Bundle/ApiBundle/Validator/Constraints/AddingEligibleProductVariantToCartValidator.php

```diff
@@ -107,4 +111,15 @@ public function validate($value, Constraint $constraint)
             );
         }
     }
+
+    private function getExistingCartItemQuantityFromCart(OrderInterface $cart, ProductVariantInterface $productVariant): int
+    {
+        foreach ($cart->getItems() as $existingCartItem) {
```

Comment:
I'm wondering, maybe we can use `OrderInterface::getItemUnitsByVariant($variant)` there? After calling `->count()` method we would have the number of item units for a specific variant - which indeed is a _cart item quantity_ 💃 Maybe we would even be able to resign from the additional private function:

```php
if (!$this-&gt;availabilityChecker-&gt;isStockSufficient(
    $productVariant,
    $value->quantity + $cart->getItemUnitsByVariant($variant)->count()
)) {
```